resolwe/flow/utils/purge.py

```diff
@@ -97,57 +98,52 @@ def subfiles(root):
     return set([os.path.join(root, filename) for filename in unreferenced_files])
 
 
-def data_purge(data_ids=None, delete=False, verbosity=0):
-    """Print files not referenced from meta data.
-
-    If data_ids not given, run on all data objects.
-    If delete is True, delete unreferenced files.
+def data_purge(location_id, delete=False, verbosity=0):
+    """Print and conditionally delete files not referenced by meta data.
 
+    :param location_id: Id of the
+        :class:`~resolwe.flow.models.DataLocation` data objects
+        reference to.
+    :param delete: If ``True``, then delete unreferenced files.
     """
-    data_path = settings.FLOW_EXECUTOR['DATA_DIR']
-    runtime_path = settings.FLOW_EXECUTOR['RUNTIME_DIR']
-    unreferenced_files = set()
+    try:
+        location = DataLocation.objects.get(id=location_id)
+    except DataLocation.DoesNotExist:
+        logger.warning("Data location does not exist", extra={'location_id': location_id})
+        return
 
-    data_qs = Data.objects.filter(status__in=[Data.STATUS_DONE, Data.STATUS_ERROR])
-    if data_ids is not None:
-        data_qs = data_qs.filter(pk__in=data_ids)
-
-    for data in data_qs:
-        root = os.path.join(data_path, str(data.id))
-
-        unreferenced_files.update(get_purge_files(
-            root,
-            data.output,
-            data.process.output_schema,
-            data.descriptor,
-            getattr(data.descriptor_schema, 'schema', [])
-        ))
-
-    # Remove any folders, which do not belong to any data objects.
-    if data_ids is None:
-        for base_path in (data_path, runtime_path):
-            for directory in os.listdir(base_path):
-                directory_path = os.path.join(base_path, directory)
-                if not os.path.isdir(directory_path):
-                    continue
-
-                try:
-                    data_id = int(directory)
-                except ValueError:
-                    continue
-
-                # Check if a data object with the given identifier exists.
-                if not Data.objects.filter(pk=data_id).exists():
-                    unreferenced_files.add(directory_path)
+    unreferenced_files = set()
+    is_location_referenced_by_data = location.data.exists()
```

Comment:
The `is_location_referenced_by_data` is used in two places and longer names adds to readability. No?

Reply:
Longer, not too long :). But each variable also adds some overhead as you have to figure out where it comes from. In this case i would argue to use `location.data.exists()` in both cases, especially as it is shorter as `is_location_referenced_by_data`.

Reply:
Using a variable with a meaningful name saves you from writing comments. You also pay extra attention when changing functionality because it forces you look where that variable is used.

Reply:
For me `location.data.exists()` is more meaningful than `is_location_referenced_by_data` if I only see a chunk of code.
I won't argue to remove it if you prefer it this way (it saves one db query, so it is better after all). But please find a shorter name :)